src/app/modules/courses/components/course-card/course-card.component.ts

```diff
@@ -15,4 +21,30 @@ export class CourseCardComponent {
   @Output() delete = new EventEmitter<number>();
 
   starIcon = faStar;
+
+  constructor(
+    private coursesService: CoursesService,
+    private dialogService: Dialog
+  ) {
+  }
+
+  deleteCourse(id: number) {
+    this.coursesService.getCourse(id)
+      .pipe(
+        switchMap(course => this.showDialog(course.name)),
+        untilDestroyed(this)
+      ).subscribe(() => this.delete.emit(id));
```

Comment:
This emit will work when user clicks confirm? Or in any case (after dialog close)?

Reply:
show dialog will return me only true result, there is a filter, but i have refactored, and add filter to delete course now